src/item.cpp

```diff
@@ -11252,7 +11252,7 @@ bool item::burn( fire_data &frd )
 {
     float burn_added = simulate_burn( frd );
 
-    if( burn_added <= 0 ) {
+    if( burn_added <= 0 || has_flag( flag_UNBREAKABLE ) ) {
```

Comment:
this check should be split off and pulled up into first line of the function because simulate_burn() mutates `frd` struct intensity/smoke etc for stuff that wasn't actually burnt

Reply:
Resolved